File: Difficulty/Easy/[125] Valid Palindrome/solution.cpp
/*
 * @lc app=leetcode id=125 lang=cpp
 *
 * [125] Valid Palindrome
 */

// @lc code=start
class Solution {
public:
    bool isPalindrome(string s) {
        int left = 0, right = s.size() - 1;
        while (left < right) {
            if (s[left] < '0' || (s[left] > '9' && s[left] < 'A') || (s[left] > 'Z' && s[left] < 'a') || s[left] > 'z') {
                left++;
                continue;                
            }
            if (s[right] < '0' || (s[right] > '9' && s[right] < 'A') || (s[right] > 'Z' && s[right] < 'a') || s[right] > 'z') {
                right--;
                continue;                
            }
            if (tolower(s[left]) == tolower(s[right])) {
                left++;
                right--;
            }else{
                return false;
            }
        }
        return true; 
    }
};
// @lc code=end

File: Difficulty/Easy/[125] Valid Palindrome/solution.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        left, right = 0, len(s) - 1
        while left < right:
            if ord(s[left]) < ord('0') or (ord(s[left]) > ord('9') and ord(s[left]) < ord('A')) or (ord(s[left]) > ord('Z') and ord(s[left]) < ord('a')) or ord(s[left]) > ord('z'):
               left += 1
               continue
            if ord(s[right]) < ord('0') or (ord(s[right]) > ord('9') and ord(s[right]) < ord('A')) or (ord(s[right]) > ord('Z') and ord(s[right]) < ord('a')) or ord(s[right]) > ord('z'):
               right -= 1
               continue
            if s[left].lower() == s[right].lower():
                left += 1
                right -= 1
            else:
                return False
        return True
                